packages/firestore/test/util/helpers.ts

```diff
@@ -1073,3 +1074,40 @@ export function computeCombinations<T>(input: T[]): T[][] {
   };
   return computeNonEmptyCombinations(input).concat([[]]);
 }
+
+/**
+ * Helper method to generate bloom filter proto value for mocking watch
+ * filter response.
+ *
+ * If the return value is null, please increase the number of bits and/or
+ * hashCount to generate a bloom filter that meets expectation.
+ */
+export function generateBloomFilterProto(config: {
+  contains: string[];
+  notContains: string[];
+  hashCount: number;
+  numOfBits: number;
+}): api.BloomFilter {
+  const DOCUMENT_PREFIX =
+    'projects/test-project/databases/(default)/documents/';
+
+  const { contains, notContains, hashCount, numOfBits } = config;
+
+  const bloomFilter = BloomFilter.create(
+    numOfBits,
+    hashCount,
+    contains.map(item => DOCUMENT_PREFIX + item)
+  );
+
+  notContains.forEach(item => {
+    if (bloomFilter.mightContain(DOCUMENT_PREFIX + item)) {
+      throw new Error(
+        'Cannot generate desired bloom filter. Please adjust the hashCount and/or number of bits.'
+      );
+    }
+  });
+  return {
+    bits: { bitmap: bloomFilter.getBitMap(), padding: bloomFilter!.padding },
```

Comment:
I don't think that the exclamation marks in `bloomFilter!` are required anymore. Please remove them if they are not.